tests/func/test_api.py

```diff
@@ -126,3 +127,29 @@ def test_open_not_cached(dvc):
     os.remove(metric_file)
     with pytest.raises(FileMissingError):
         api.read(metric_file)
+
+
+def test_summon(tmp_dir, erepo_dir, dvc):
+    grimoire_yaml = ruamel.yaml.dump(
+        {
+            "spells": [
+                {
+                    "name": "three",
+                    "description": "The sum of 1 + 2",
+                    "file": "calculator.py",
+                    "method": "sum",
+                    "params": {"x": 1, "y": 2},
+                    "deps": ["foo"],
```

Comment:
`file`, `method`, `params ` are python specific. How about custom types?

Reply:
In the specification repos (`dvc-pretrained-resnet` and `df_sea_ice`) these type-specific options were separated to a dedicated section `summon` with mandatory `type`. It was defined in such a way to abstract out objects/artifacts from types.

Reply:
@dmpetrov , didn't understand what was the need for a `summon` key. Are there artifacts that can't be summoned?

Reply:
@mroutis it might a data catalog theoretically without summons, also we need to isolate `summon` subfields from the rest to only  reserve `summon`.

Reply:
Got it :+1: